### ISSUE TITLE
CA2000 (DisposeObjectsBeforeLosingScope) false positive with async/await

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#2212]
#### Analyzer package

Example: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

Example: 2.9.0-beta2.19165.1+b2ba91d7

#### Diagnostic ID

Example: [CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope)

#### Repro steps

```csharp
using System;
using System.Threading.Tasks;

class C : IDisposable
{
    public void Dispose()
    {
    }

    public Task<C> M1_Task()
    {
        return Task.FromResult(new C());
    }

    public async Task M2_Task()
    {
        var c = await M1_Task().ConfigureAwait(false);
    }
}
```

#### Expected behavior
Single CA2000 on `await M1_Task().ConfigureAwait(false)`

#### Actual behavior
Warning	CA2000	In method 'Task C.M2_Task()', call System.IDisposable.Dispose on object created by 'new C()' before all references to it are out of scope.	12	32	
Warning	CA2000	In method 'Task<C> C.M1_Task()', call System.IDisposable.Dispose on object created by 'new C()' before all references to it are out of scope.	12	32
